src/DefaultBuilder/src/WebApplication.cs

```diff
@@ -237,4 +241,39 @@ private void Listen(string? url)
         addresses.Clear();
         addresses.Add(url);
     }
+
+    private string DebuggerToString()
+    {
+        return $@"ApplicationName = ""{Environment.ApplicationName}"", IsRunning = {(IsRunning ? "true" : "false")}";
+    }
+
+    // Web app is running if the app has been started and hasn't been stopped.
+    private bool IsRunning => Lifetime.ApplicationStarted.IsCancellationRequested && !Lifetime.ApplicationStopped.IsCancellationRequested;
+
+    internal sealed class WebApplicationDebugView(WebApplication webApplication)
+    {
+        private readonly WebApplication _webApplication = webApplication;
+
+        public IServiceProvider Services => _webApplication.Services;
+        public IConfiguration Configuration => _webApplication.Configuration;
+        public IWebHostEnvironment Environment => _webApplication.Environment;
+        public IHostApplicationLifetime Lifetime => _webApplication.Lifetime;
+        public ILogger Logger => _webApplication.Logger;
+        public ICollection<string> Urls => _webApplication.Urls;
```

Comment:
The list should be short and readable, display it directly.
```suggestion
        public string Urls => string.Join(", " _webApplication.Urls);
```